tests/Tests/ORM/Functional/Ticket/GH11501Test.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Exception\ORMException;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH11501Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH11501AbstractTestEntity::class,
+            GH11501TestEntityOne::class,
+            GH11501TestEntityTwo::class,
+            GH11501TestEntityHolder::class,
+        ]);
+    }
+
+    /**
+     * @throws ORMException
+     */
+    public function testDeleteOneToManyCollectionWithSingleTableInheritance(): void
+    {
+        $testEntityOne    = new GH11501TestEntityOne();
+        $testEntityTwo    = new GH11501TestEntityTwo();
+        $testEntityHolder = new GH11501TestEntityHolder();
+
+        $testEntityOne->testEntityHolder = $testEntityHolder;
+        $testEntityHolder->testEntities->add($testEntityOne);
+
+        $testEntityTwo->testEntityHolder = $testEntityHolder;
+        $testEntityHolder->testEntities->add($testEntityTwo);
+
+        $em = $this->getEntityManager();
+        $em->persist($testEntityOne);
+        $em->persist($testEntityTwo);
+        $em->persist($testEntityHolder);
+        $em->flush();
+
+        $testEntityHolder->testEntities = new ArrayCollection();
```

Comment:
Here, you are only changing the inverse side of the relationship, right? With that in mind, and after reading [this](https://www.doctrine-project.org/projects/doctrine-orm/en/3.2/reference/unitofwork-associations.html), I don't see why "All records should have been deleted". Am I missing something?

Reply:
Maybe.. unless I'm misunderstanding, I think that linked page is incomplete as it doesn't account for orhpanRemoval. [This](https://www.doctrine-project.org/projects/doctrine-orm/en/3.2/reference/working-with-associations.html#orphan-removal) suggests the inverse side is not ignored?

Reply:
It looks like you're correct. Maybe that first page should be updated to account for that exception to the general rule?